sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/CatchRethrow.cs

```diff
@@ -116,7 +116,7 @@ private void RaiseOnInvalidCatch(SyntaxNodeAnalysisContext context)
             return catches
                 .Select(clause =>
                     clause.Declaration?.Type != null
-                        ? semanticModel.GetTypeInfo(clause.Declaration?.Type).Type as INamedTypeSymbol
+                        ? semanticModel.GetTypeInfo(clause.Declaration.Type).Type as INamedTypeSymbol

```

Comment:
Because we test `clause.Declaration?.Type != null` before so when we reach this branch we know that `clause.Declaration.Type` isn't null (and so `clause.DeclarationType` isn't).